### PR TITLE
Improve Managing Secrets using a Configuration File

### PR DESCRIPTION
**Pull request by shannonxtreme**
Umbrella issue: #4491
Replaces #34856

This PR makes a series of updates to the Managing Secrets using a Configuration File topic:

- Improve wording for accuracy, especially related to referring to API object definition and creation
- Use the imperative form for steps
- Add numbers to steps
- Consolidate info about stringdata field away from the task flow
- Replace the verify step, which uses kubectl, with a link to the same steps in the Managing Secrets using kubectl topic

Reviewers: please review each of the six commits. I haven't squashed on purpose, to split the changes into independent chunks.

/sig docs
/language en

### CODE REVIEW

File: content/en/docs/tasks/configmap-secret/managing-secret-using-config-file.md [hunk 1]
```diff
@@ -15,63 +15,67 @@ description: Creating Secret objects using resource configuration file.
...
+   echo -n 'admin' | base64
+   echo -n '1f2d1e2e67df' | base64
+   ```
 
-```
-YWRtaW4=
-```
+   {{< note >}}
+   The serialized JSON and YAML values of Secret data are encoded as base64
+   strings. Newlines are not valid within these strings and must be omitted. When using the `base64` utility on Darwin/macOS, users should avoid using the `-b` option to split long lines. Conversely, Linux users *should* add the option `-w 0` to `base64` commands or the pipeline `base64 | tr -d '\n'` if the `-w` option is not available. {{< /note >}}
```
Comment (divya-mohan0209) [inline content/en/docs/tasks/configmap-secret/managing-secret-using-config-file.md:38]: Nit: Wrapping the text properly
```suggestion
   strings. Newlines are not valid within these strings and must be omitted. 
   When using the `base64` utility on Darwin/macOS, users should avoid 
   using the `-b` option to split long lines. Conversely, Linux users *should* 
   add the option `-w 0` to `base64` commands or the pipeline `base64 | tr -d '\n'` 
   if the `-w` option is not available. 
   {{< /note >}}
```

File: content/en/docs/tasks/configmap-secret/managing-secret-using-kubectl.md [hunk 2]
```diff
@@ -111,7 +111,7 @@ username:    5 bytes
 
 The commands `kubectl get` and `kubectl describe` avoid showing the contents
 of a `Secret` by default. This is to protect the `Secret` from being exposed
-accidentally, or from being stored in a terminal log.
+accidentally, or from being stored in a terminal log. To check the actual content of the encoded data, please refer to [Decoding the Secret](#decoding-secret).
```
Comment (divya-mohan0209) [inline content/en/docs/tasks/configmap-secret/managing-secret-using-kubectl.md:114]:
```suggestion
accidentally, or from being stored in a terminal log. 
To check the actual content of the encoded data, please refer to [Decoding the Secret](#decoding-secret).
```

File: content/en/docs/tasks/configmap-secret/managing-secret-using-config-file.md [hunk 3]
```diff
@@ -143,14 +132,9 @@ metadata:
...
-The commands `kubectl get` and `kubectl describe` avoid showing the contents of a `Secret` by
-default. This is to protect the `Secret` from being exposed accidentally to an onlooker,
-or from being stored in a terminal log.
-To check the actual content of the encoded data, please refer to
-[decoding secret](/docs/tasks/configmap-secret/managing-secret-using-kubectl/#decoding-secret).
+### Specifying both `data` and `stringData`
 
-If a field, such as `username`, is specified in both `data` and `stringData`,
-the value from `stringData` is used. For example, the following Secret definition:
+If you specify a field in both `data` and `stringData`, the value from `stringData` is used. For example, if you define the following Secret:
```
Comment (divya-mohan0209) [inline content/en/docs/tasks/configmap-secret/managing-secret-using-config-file.md:137]:
```suggestion
If you specify a field in both `data` and `stringData`, the value from `stringData` is used. 
For example, if you define the following Secret:
```

**Review by divya-mohan0209 — COMMENTED** [review]
Minor text-wrapping nits. Rest looks good to me.

File: content/en/docs/tasks/configmap-secret/managing-secret-using-config-file.md [hunk 4]
```diff
@@ -15,63 +15,67 @@ description: Creating Secret objects using resource configuration file.
 
 ## Create the Config file
```
Comment (sftim) [inline content/en/docs/tasks/configmap-secret/managing-secret-using-config-file.md:16]: I recommend using the correct term in the heading and the text: “manifest” (not “config file”)

File: content/en/docs/tasks/configmap-secret/managing-secret-using-config-file.md [hunk 5]
```diff
@@ -15,63 +15,67 @@ description: Creating Secret objects using resource configuration file.
...
-The output is similar to:
+   ```
+   YWRtaW4=
+   MWYyZDFlMmU2N2Rm
+   ```
 
-```
-MWYyZDFlMmU2N2Rm
-```
+1. Create the configuration file:
```
Comment (sftim) [inline content/en/docs/tasks/configmap-secret/managing-secret-using-config-file.md:47]: “manifest”

File: content/en/docs/tasks/configmap-secret/managing-secret-using-kubectl.md [hunk 6]
```diff
@@ -113,6 +113,8 @@ The commands `kubectl get` and `kubectl describe` avoid showing the contents
 of a `Secret` by default. This is to protect the `Secret` from being exposed
 accidentally, or from being stored in a terminal log.
 
+To check the actual content of the encoded data, please refer to [Decoding the Secret](#decoding-secret).
```
Comment (sftim) [inline content/en/docs/tasks/configmap-secret/managing-secret-using-kubectl.md:116]:
```suggestion
To check the actual content of the encoded data, refer to [Decoding the Secret](#decoding-secret).
```

**Comment by sftim** [comment]
LGTM

Future work (and best saved for a different PR): change the “Decoding the Secret” section to show an alternative approach that uses `go-template` output format, and doesn't require an external `base64` tool.

**Review by divya-mohan0209 — COMMENTED** [review]
LGTM from me!

**Comment by sftim** [comment]
Taking LGTM from https://github.com/kubernetes/website/pull/35383#pullrequestreview-1053483710

/lgtm

**Pull request merged**